polaris/ocean/tasks/barotropic_gyre/analysis.py

```diff
@@ -61,26 +62,35 @@ def run(self):
                                    boundary_condition=self.boundary_condition)
         print(f'L2 error norm for {self.boundary_condition} bsf: {error:1.2e}')
 
+        descriptor = mosaic.Descriptor(ds_mesh)
+
         use_mplstyle()
         pad = 20
         fig, axes = plt.subplots(nrows=1, ncols=3, figsize=(12, 2))
-        x0 = ds_mesh.xEdge.min()
-        y0 = ds_mesh.yEdge.min()
-        x_vertex = (ds_mesh['xVertex'] - x0) * 1.e-3
-        y_vertex = (ds_mesh['yVertex'] - y0) * 1.e-3
+        x0 = float(ds_mesh.xEdge.min())
+        y0 = float(ds_mesh.yEdge.min())
```

Comment:
I think the preferred syntax is:
```suggestion
        x0 = ds_mesh.xEdge.min().values
        y0 = ds_mesh.yEdge.min().values
```
Let me know if you've seen otherwise.

Reply:
Also, I would have thought the `xarray.DataArray` would have worked but maybe `descriptor.vertex_patches` is a` numpy` array, not an `xarray.DataArray`?

Reply:
Yeah `descriptor.vertex_patches` is a `numpy` arrray, not an `xarray.DataArray`. So using the `xarray.DataArray` returned by min/max will cause broadcasting errors. 

My personal preference has always been to cast the min/max as a float, since there can only ever be one return value it's safe, and that way it's actually a float. Using `.values` will return a zero dimensional `numpy` array, which is also fine. Broadcasting will work correctly, it's just the type of the return is not what you'd necessarily expect.  

I'm happy to switch to `.values`.

Reply:
Okay, stick with what you have.

Reply:
I see you already changed it, maybe easier to leave it now.